src/printer_tree.c

```diff
@@ -0,0 +1,3107 @@
+{
+    /* +--<node>
+     *    +--<node>
+     */
+    wr.actual_pos++;
+    return wr;
+}
+
+/**
+ * @brief Setting '|' symbol because node is divided or it is not last sibling.
+ * @param[in] wr is source of wrapper.
+ * @return New wrapper which is marked at actual position and shifted.
+ */
+static struct trt_wrapper
+trp_wrapper_set_mark(struct trt_wrapper wr)
+{
+    wr.bit_marks1 |= 1U << wr.actual_pos;
+    return trp_wrapper_set_shift(wr);
+}
+
+/**
+ * @brief Setting ' ' symbol if node is last sibling otherwise set '|'.
+ * @param[in] wr is actual wrapper.
+ * @param[in] last_one is flag. Value 1 saying if the node is the last and has no more siblings.
+ * @return New wrapper for the actual node.
+ */
+static struct trt_wrapper
+trp_wrapper_if_last_sibling(struct trt_wrapper wr, ly_bool last_one)
+{
+    return last_one ? trp_wrapper_set_shift(wr) : trp_wrapper_set_mark(wr);
+}
+
+/**
+ * @brief Test if the wrappers are equivalent.
+ * @param[in] first is the first wrapper.
+ * @param[in] second is the second wrapper.
+ * @return 1 if the wrappers are equivalent otherwise 0.
+ */
+static ly_bool
+trp_wrapper_eq(struct trt_wrapper first, struct trt_wrapper second)
+{
+    const ly_bool a = first.type == second.type;
+    const ly_bool b = first.bit_marks1 == second.bit_marks1;
+    const ly_bool c = first.actual_pos == second.actual_pos;
+
+    return a && b && c;
+}
+
+/**
+ * @brief Print "  |  " sequence on line.
+ * @param[in] wr is wrapper to be printed.
+ * @param[in,out] out is output handler.
+ */
+static void
+trp_print_wrapper(struct trt_wrapper wr, struct ly_out *out)
+{
+    {
+        uint32_t lb;
+        if (wr.type == TRD_WRAPPER_TOP) {
+            lb = TRD_INDENT_LINE_BEGIN;
+        } else if (wr.type == TRD_WRAPPER_BODY) {
+            lb = TRD_INDENT_LINE_BEGIN * 2;
+        } else {
+            lb = TRD_INDENT_LINE_BEGIN;
+        }
+
+        ly_print_(out, "%*c", lb, ' ');
+    }
+
+    if (trp_wrapper_eq(wr, trp_init_wrapper_top())) {
+        return;
+    }
+
+    for (uint32_t i = 0; i < wr.actual_pos; i++) {
+        /** Test if the bit on the index is set. */
+        if ((wr.bit_marks1 >> i) & 1U) {
+            ly_print_(out, "|");
+        } else {
+            ly_print_(out, " ");
+        }
+
+        if (i != wr.actual_pos) {
+            ly_print_(out, "%*c", TRD_INDENT_BTW_SIBLINGS, ' ');
+        }
+    }
+}
+
+/**
+ * @brief Create struct trt_node_name as empty.
+ */
+static struct trt_node_name
+trp_empty_node_name()
+{
+    return (struct trt_node_name) {
+               TRD_NODE_ELSE,
+               NULL,
+               NULL
+    };
+}
+
+/**
+ * @brief Check if struct trt_node_name is empty.
+ * @param[in] node_name is item to test.
+ */
+static ly_bool
+trp_node_name_is_empty(struct trt_node_name node_name)
+{
+    return node_name.str == NULL;
+}
+
+/**
+ * @brief Create empty struct trt_type.
+ */
+static struct trt_type
+trp_empty_type()
+{
+    return (struct trt_type) {
```

Comment:
An interesting construct but I do not like it. It is probably just a personal preference, but I would expect this to simply be a macro (the code in `{}`). Just like `PTHREAD_MUTEX_INITIALIZER`, for example, any static initialization.